tools/wandb_export_history.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+"""Export W&B run history.
+
+This module uses the W&B public api to download unsampled history to a
+sqlite database.
+
+Usage:
+    ./wandb_export_history --run entity/project/run_id --db_file save.db
+
+or:
+    ```python
+    import wandb_export_history
+    wandb_export_history(run="entity/project/run_id")
+    ```
+
+"""
+
+import argparse
+import sqlite3
+from typing import Dict, Iterator, List
+
+import pandas as pd  # type: ignore
+import wandb
+
+
+DB_FILE = "run.db"
+
+
+def chunk(n: int, iterable) -> Iterator[List[Dict]]:
+    done = False
+    while not done:
+        data = []
+        try:
+            for _ in range(n):
+                data.append(next(iterable))
+        except StopIteration:
+            if data:
+                yield data
+            # done = True
+            break
+        yield data
+
+
+def wandb_export_history(
+    *,
+    run,
+    api=None,
+    db_file=None,
+    db_table=None,
+    db_replace: bool = None,
+    history_exclude_prefix=None,
+    read_page_size=None,
+    write_page_size=None,
+) -> int:
+    api = api or wandb.Api()
+    db_file = db_file or DB_FILE
+    db_table = db_table or "history"
+    history_exclude_prefix = history_exclude_prefix or "system/"
+    read_page_size = read_page_size or 1000
+    write_page_size = write_page_size or 1000
+
+    run = api.run(run)
+    keys = run.history_keys.get("keys", [])
+    if history_exclude_prefix:
+        keys = list(filter(lambda x: not x.startswith(history_exclude_prefix), keys))
+
+    db_file = db_file
+    db = sqlite3.connect(db_file)
+
+    history = run.scan_history(page_size=read_page_size)
```

Comment:
i think you can use `keys` to scan only history for these keys?

Reply:
We can't specify keys or we will get sampled:
```
        if keys is None:
            return HistoryScan(
                run=self,
                client=self.client,
                page_size=page_size,
                min_step=min_step,
                max_step=max_step,
            )
        else:
            return SampledHistoryScan(
```

Reply:
Yeah i saw that one, but I assumed Sampled meant that it's not all keys... this api seems weird